sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala

```diff
@@ -437,6 +437,12 @@ trait V2CreateTablePlan extends LogicalPlan {
   def withPartitioning(rewritten: Seq[Transform]): V2CreateTablePlan
 }
 
+/** A trait used for logical plan nodes that create or replace new table specifications. */
+trait HasTableSpec extends LogicalPlan {
+  def tableSpec: TableSpec
+  def unresolvedOptionsList: UnresolvedOptionsList
```

Comment:
Even when a logical plan like `CreateTable` is resolved, it will carry the method `unresolvedOptionsList`

Reply:
Good point, I renamed this to `optionsListExpressions` to clarify.